update-common/src/artifacts/update_plan.rs

```diff
@@ -317,6 +321,50 @@ impl<'a> UpdatePlanBuilder<'a> {
             },
         )?;
 
+        // We need to get all the signing information now to properly check
+        // version at builder time (builder time is not async)
+        let image_a = rot_a_data
+            .reader_stream()
+            .and_then(|stream| async {
+                let mut buf = Vec::with_capacity(rot_a_data.file_size());
+                tokio_util::io::StreamReader::new(stream)
+                    .read_to_end(&mut buf)
+                    .await
+                    .unwrap();
```

Comment:
These are extracted as files on disk, so I believe they can produce errors. Even if they can't, it would be good to replace the `unwrap()` with an `expect("justification")`.

Reply:
wicketd mentions this can be an I/O error https://github.com/oxidecomputer/omicron/blob/main/wicketd/src/update_tracker.rs#L1690 . I'll match that here.